tests/run-make/c-link-to-rust-dylib/rmake.rs

```diff
@@ -0,0 +1,44 @@
+// This test checks that C linking with Rust does not encounter any errors, with dynamic libraries.
+// See <https://github.com/rust-lang/rust/issues/10434>.
+
+//@ ignore-cross-compile
+
+use std::fs::remove_file;
+
+use run_make_support::{
+    cc, dynamic_lib_extension, is_msvc, read_dir, run, run_fail, rustc, tmp_dir,
+};
+
+fn main() {
+    rustc().input("foo.rs").run();
+
+    if is_msvc() {
+        let lib = tmp_dir().join("foo.dll.lib");
+
+        // We remove the file in case it exists.
+        let _ = remove_file(&lib);
```

Comment:
Problem: I don't think the removal here is correct. We need the dynamic lib here still for `cc` to use.